feat(aio): marketing design refresh

sjtrimble opened the pull request:
home page and sidenav layout and styling

**Please check if the PR fulfills these requirements**
- [x] The commit message follows our guidelines: https://github.com/angular/angular/blob/master/CONTRIBUTING.md#commit
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)


**What kind of change does this PR introduce?** (check one with "x")
```
[ ] Bugfix
[x] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Build related changes
[ ] CI related changes
[ ] Other... Please describe:
```

**What is the current behavior?** (You can also link to an open issue here)



**What is the new behavior?**



**Does this PR introduce a breaking change?** (check one with "x")
```
[ ] Yes
[x] No
```

If this PR contains a breaking change, please describe the impact and migration path for existing applications: ...


**Other information**:

mary-poppins commented:
The angular.io preview for 88649919f9f4c419863e8b2d4352dc641105fe8b is available [here][1].

[1]: https://pr16228-88649919f9f4c419863e8b2d4352dc641105fe8b.ngbuilds.io/

mary-poppins commented:
The angular.io preview for 86cf43406b6edaf786e9f37cf19d6e3f0ea17fcb is available [here][1].

[1]: https://pr16228-86cf43406b6edaf786e9f37cf19d6e3f0ea17fcb.ngbuilds.io/

mary-poppins commented:
The angular.io preview for 99be689b1596ec0ee778754d67a919bb867c8e5e is available [here][1].

[1]: https://pr16228-99be689b1596ec0ee778754d67a919bb867c8e5e.ngbuilds.io/

mary-poppins commented:
The angular.io preview for 5377386fe7ef36dde659e691e31805c4514b085b is available [here][1].

[1]: https://pr16228-5377386fe7ef36dde659e691e31805c4514b085b.ngbuilds.io/

mary-poppins commented:
The angular.io preview for 7b51039b4f2f2f642d9f0f6195cafed2e7fdf1ae is available [here][1].

[1]: https://pr16228-7b51039b4f2f2f642d9f0f6195cafed2e7fdf1ae.ngbuilds.io/

mary-poppins commented:
The angular.io preview for 4d6da37e06f76de4118aed450eb80c2a9ca6ecc8 is available [here][1].

[1]: https://pr16228-4d6da37e06f76de4118aed450eb80c2a9ca6ecc8.ngbuilds.io/

mary-poppins commented:
The angular.io preview for 8ab699914a8bd31953214cc64713742cb89159a2 is available [here][1].

[1]: https://pr16228-8ab699914a8bd31953214cc64713742cb89159a2.ngbuilds.io/

mary-poppins commented:
The angular.io preview for 445725d997221d39f781e1f5bf45692455fe38ab is available [here][1].

[1]: https://pr16228-445725d997221d39f781e1f5bf45692455fe38ab.ngbuilds.io/

wardbell commented:
@sjtrimble I rebased for you, fixing a few things (like disabling the TOC for this page by adding the "no-toc" class to the `<h1>`.

The hamburger no longer displays although a big square invisible button is there in its place. Is that intentional?

mary-poppins commented:
The angular.io preview for d5dc6826f18e12f5869ed95e41e32aeaf7c0ff88 is available [here][1].

[1]: https://pr16228-d5dc6826f18e12f5869ed95e41e32aeaf7c0ff88.ngbuilds.io/

mary-poppins commented:
The angular.io preview for 425cbc80d36296638cd6ab97244d2b6da230b5fe is available [here][1].

[1]: https://pr16228-425cbc80d36296638cd6ab97244d2b6da230b5fe.ngbuilds.io/

mary-poppins commented:
The angular.io preview for d67de863401fd55bde4d31aa95de43d77f40c457 is available [here][1].

[1]: https://pr16228-d67de863401fd55bde4d31aa95de43d77f40c457.ngbuilds.io/

mary-poppins commented:
The angular.io preview for dfcc6601be2de89699c5a50e99de35d17ea0fde7 is available [here][1].

[1]: https://pr16228-dfcc6601be2de89699c5a50e99de35d17ea0fde7.ngbuilds.io/

mary-poppins commented:
The angular.io preview for b7f6f285a144d8d4e1bbb1ecf98c0ba031aa771d is available [here][1].

[1]: https://pr16228-b7f6f285a144d8d4e1bbb1ecf98c0ba031aa771d.ngbuilds.io/

mary-poppins commented:
The angular.io preview for cdc3dd9cda2d25b97aa23263d03c189b2826fab6 is available [here][1].

[1]: https://pr16228-cdc3dd9cda2d25b97aa23263d03c189b2826fab6.ngbuilds.io/

mary-poppins commented:
The angular.io preview for ebae17e27da7af162af60a00032bb334bc24899f is available [here][1].

[1]: https://pr16228-ebae17e27da7af162af60a00032bb334bc24899f.ngbuilds.io/

juleskremer commented:
[Features Page]: please remove the "top header" that says features & benefits, not needed;
[Homepage]:  something is messed up; please move the A logo slightly higher and note that the bottom of the page the CSS is messed up :)

mary-poppins commented:
The angular.io preview for 15af46f31ccc3ad7d16d7cf51e0e6806179794b1 is available [here][1].

[1]: https://pr16228-15af46f31ccc3ad7d16d7cf51e0e6806179794b1.ngbuilds.io/

mary-poppins commented:
The angular.io preview for e6a5819bf5cc339d553a81b34f5ae4d1c35458cc is available [here][1].

[1]: https://pr16228-e6a5819bf5cc339d553a81b34f5ae4d1c35458cc.ngbuilds.io/

mary-poppins commented:
The angular.io preview for 44f6e22b1560b355a38381e6d51c615349ec2bc1 is available [here][1].

[1]: https://pr16228-44f6e22b1560b355a38381e6d51c615349ec2bc1.ngbuilds.io/

mary-poppins commented:
The angular.io preview for 33c5b1e31fb773be4c50edad89fbc46d055d0e0d is available [here][1].

[1]: https://pr16228-33c5b1e31fb773be4c50edad89fbc46d055d0e0d.ngbuilds.io/

mary-poppins commented:
The angular.io preview for 365dfb084f2dac9f77688b2170584ed82f404143 is available [here][1].

[1]: https://pr16228-365dfb084f2dac9f77688b2170584ed82f404143.ngbuilds.io/

mary-poppins commented:
The angular.io preview for 0d6ec0ccfffc4593d7542742bf847d9fe56f7965 is available [here][1].

[1]: https://pr16228-0d6ec0ccfffc4593d7542742bf847d9fe56f7965.ngbuilds.io/

mary-poppins commented:
The angular.io preview for 34f0ab305125808a983011deb4c32873b9a48457 is available [here][1].

[1]: https://pr16228-34f0ab305125808a983011deb4c32873b9a48457.ngbuilds.io/

mary-poppins commented:
The angular.io preview for e247938b5e37dcc123a2cc5a6f30eb77cd5e78b3 is available [here][1].

[1]: https://pr16228-e247938b5e37dcc123a2cc5a6f30eb77cd5e78b3.ngbuilds.io/

mary-poppins commented:
The angular.io preview for c9139c8f32901a12eb33777b527279eab26dcd6e is available [here][1].

[1]: https://pr16228-c9139c8f32901a12eb33777b527279eab26dcd6e.ngbuilds.io/

petebacondarwin commented:
@sjtrimble can you rebase and fix the conflicts please

sjtrimble commented:
Yes, will do today.

mary-poppins commented:
The angular.io preview for 44b1a3fc8ba4a911c10d3875c42dd21146d34317 is available [here][1].

[1]: https://pr16228-44b1a3fc8ba4a911c10d3875c42dd21146d34317.ngbuilds.io/

petebacondarwin commented:
The reason there are hundreds of files is because the "old" `src/content` folder was committed. This was a generated folder and so should not be there. Please can you remove it - and rebase to resolve the commit.

petebacondarwin commented:
I have removed the generated files and force-pushed

petebacondarwin commented on aio/content/marketing/features.html:
```diff
@@ -1,88 +1,105 @@
+      <div class="feature-section">
+        <div class="feature-header">
+          <h2 class="text-headline">Cross Platform</h2>
+          <img src="../assets/images/icons/feature-icon.svg" height="70px">
```
we are now keeping content specific images in the `aio/content/images` folder. I suggest that we move these to a `aio/content/images/marketing` folder.

petebacondarwin commented on aio/src/styles/2-modules/_hero.scss:
```diff
@@ -110,4 +123,4 @@
   }
-}
\ No newline at end of file
+}  
```
trailing whitespace 😱

petebacondarwin commented on aio/src/styles/2-modules/_hero.scss:
```diff
@@ -86,7 +96,10 @@
+      width: 400px; // SJT CHANGED FROM 240px
+      position: absolute; // SJT ADD
+      top: 60px; // SJT ADD
+      left: 150px; // SJT ADD
```
Are these `SJT` comments just for your use while working on this @sjtrimble ?
I don't think they should stay here, we have git for tracking what changed and who changed it.

petebacondarwin commented on aio/src/styles/2-modules/_hero.scss:
```diff
@@ -12,17 +12,24 @@
+  overflow: hidden;
+  transform: skewY(8deg);
+  transform-origin: 100%;
+  // 
```
random empty comment?

petebacondarwin commented on aio/src/styles/2-modules/_announcement-bar.scss:
```diff
@@ -1,180 +1,180 @@
-  }
-}
\ No newline at end of file
+// /* Announcment Bar Module */
```
How about we just delete this file?

mary-poppins commented:
The angular.io preview for 3d4cce9af33fe45cb5bdd3a475a9326d04d1c77f is available [here][1].

[1]: https://pr16228-3d4cce9af33fe45cb5bdd3a475a9326d04d1c77f.ngbuilds.io/

petebacondarwin commented:
The search box looks much nicer but there are some weird corner cases that need attention...

When the screen is iphone 5 width the search box looks jammed against the logo:
![screen shot 2017-05-24 at 10 21 49](https://cloud.githubusercontent.com/assets/15655/26396363/eb78a65c-406a-11e7-818a-b4454b4ef667.png)

Between 480px and 481px width the search box become narrower, but doesn't then exhibit the expand on click behaviour:
![screen shot 2017-05-24 at 10 23 29](https://cloud.githubusercontent.com/assets/15655/26396456/31ca6384-406b-11e7-88c2-91bf01ad7a80.png)


![screen shot 2017-05-24 at 10 24 51](https://cloud.githubusercontent.com/assets/15655/26396491/42bf550a-406b-11e7-8f37-15329b2e38b5.png)

petebacondarwin commented:
There is slight strange behaviour on the home page, when the screen is 600px or less. The top bar becomes white:
![screen shot 2017-05-24 at 10 26 54](https://cloud.githubusercontent.com/assets/15655/26396581/8a4f620c-406b-11e7-88bb-ae8b68eb790f.png)
![screen shot 2017-05-24 at 10 27 19](https://cloud.githubusercontent.com/assets/15655/26396605/9a632eda-406b-11e7-8f17-4a07c3399d56.png)

This is not the case for the other pages, whose TobBar remains blue.

petebacondarwin commented:
In narrow screens the top h1 headings are now hidden by the topbar.
![screen shot 2017-05-24 at 10 30 01](https://cloud.githubusercontent.com/assets/15655/26396760/0e80cf84-406c-11e7-9afd-e6b6a311276f.png)

petebacondarwin commented:
Something weird is happening with the doc version selector. When the browser window is flush against the left hand side of my right hand monitor, clicking on the dropdown makes the list appear on my left hand monitor.
![screen shot 2017-05-24 at 10 33 22](https://cloud.githubusercontent.com/assets/15655/26396939/958299fe-406c-11e7-9dc7-4978032ff8f7.png)

This doesn't happen on the staging site and it doesn't happen if there is some space to the left of the browser.  I think the problem is that the list is now being positioned slightly outside the browser window:
![screen shot 2017-05-24 at 10 35 08](https://cloud.githubusercontent.com/assets/15655/26396998/c461c024-406c-11e7-9fc5-42267217fe37.png)

petebacondarwin commented on aio/src/styles/1-layouts/_sidenav.scss:
```diff
@@ -131,7 +152,7 @@ button.vertical-menu-item {
 .heading.selected.level-1,
 .heading-children.selected.level-1 {
-  border-left: 3px $blue solid;
+  // border-left: 3px $blue solid;
```
we should just delete rules if they are not being used, rather than commenting them out.

petebacondarwin commented on aio/src/styles/1-layouts/_sidenav.scss:
```diff
@@ -34,7 +60,7 @@ md-sidenav-container.sidenav-container {
     margin: 0;
     transform: none;
-    padding-top: 64px;
+    // padding-top: 64px; // SJT REMOVE
```
LOL

petebacondarwin commented on aio/src/styles/1-layouts/_content-layout.scss:
```diff
@@ -7,9 +7,14 @@
 
+aio-shell.page-docs {
+  .sidenav-content {
+    padding: 6rem 3rem 3rem 3rem; // THIS CAUSES THE TOP NAV TOOLBAR TO JUMP BETWEEN DOCS AND OTHER PAGES
```
Is this not a bad thing?

petebacondarwin commented on aio/src/app/layout/nav-item/nav-item.component.html:
```diff
@@ -23,4 +23,4 @@
     [node]="node" [selectedNodes]="selectedNodes"></aio-nav-item>
   </div>
-</div>
+</div>
```
revert this.

petebacondarwin commented on aio/src/app/app.component.ts:
```diff
@@ -61,7 +61,7 @@ export class AppComponent implements OnInit {
   private previousNavView: string;
 
-  private sideBySideWidth = 1032;
+  private sideBySideWidth = 992;
```
this is probably the cause of the unit test failures.

petebacondarwin commented on aio/content/marketing/index.html:
```diff
@@ -1,27 +1,52 @@
+        <img src="assets/images/logos/angular/angular-banner-logo-grey.png" width="64"/>
+        <p>Angular v4.0 is out! Smaller, faster, no biggie...</p>
+        <button class="button">
+          <a href="http://angularjs.blogspot.com/2017/03/angular-400-now-available.html" target="_blank">Learn More</a><span class="material-icons">arrow_forward</span>
```
Given that this is the home page and will likely be a landing page, there is a chance that using ligatures for the icons here will cause a flash of unwanted content. We should probably add this icon to the `MdIconRegistry` so that it comes preloaded.

petebacondarwin commented on aio/content/marketing/events.html:
```diff
@@ -1,7 +1,9 @@
 
-<h3>Where we'll be presenting:</h3>
 <article class="l-content ">
+<h3>Where we'll be presenting:</h3>
```
Why `h3`? From a semantic point of view this would be better as `h2`.

sjtrimble replied:
I changed this to be a paragraph. We can't use h2's without anchors ...

petebacondarwin reviewed (dismissed):
Can you move the images into the `content` folder.
We need to address the issues with the hamburger topbar background, search box and doc version picker.
The unit tests need to be fixed and we could do with all the commented out stuff being removed.

Personally, I don't like not having the hamburger on the home page but I guess that is what was requested.

mary-poppins commented:
The angular.io preview for 9b224089a2959dc78b44a56d767d2297f8e35abd is available [here][1].

[1]: https://pr16228-9b224089a2959dc78b44a56d767d2297f8e35abd.ngbuilds.io/

mary-poppins commented:
The angular.io preview for 2c9d41b274aa9a58a404cdcb690ef7401a8262c6 is available [here][1].

[1]: https://pr16228-2c9d41b274aa9a58a404cdcb690ef7401a8262c6.ngbuilds.io/

IgorMinar commented:
- [ ] `/about` the twitter and link icons are not visible in mobile mode
- [ ] `/events` content in the table is cropped off (see screenshot)
  <img width="200" alt="screen shot 2017-05-24 at 11 39 04 am" src="https://cloud.githubusercontent.com/assets/216296/26419786/ad796f20-4075-11e7-806d-72fc042313e2.png">
- [ ] `/` can you please take off the bottom clip? I talked to Jules and she prefers the bottom of the page without it (see mock screenshot #2)
<img width="200" alt="screen shot 2017-05-24 at 3 59 57 pm" src="https://cloud.githubusercontent.com/assets/216296/26429145/3c3039b4-409a-11e7-9795-3c5dfbb9b9e4.png">

sjtrimble committed:
feat(aio): marketing design refresh

- Add styling for active TOC item
- Add sidenav styles
- Change header tags to divs from index marketing page to remove anchors
- Fix use of card mixin and create separate card-docs class
- Add marketing styling
- Topnav styling when on home landing page
- Remove hamburger menu on home page
- Add fully rounded border to top nav toolbar search input
- Add mobile styles
- Add title banner to other marketing pages

mary-poppins commented:
The angular.io preview for c451d6c788df8606574bb7cd49117c31ea31a3a1 is available [here][1].

[1]: https://pr16228-c451d6c788df8606574bb7cd49117c31ea31a3a1.ngbuilds.io/

sjtrimble commented:
@IgorMinar I removed the bottom slice on the home page and fixed the events table. I vote we move the about page to a different PR as this one's focus was mainly on the marketing pages.

@petebacondarwin Are you sure you want me to move the images to the content folder? Do you want them in the marketing folder, or...? I removed the hundreds of extra files.

mary-poppins commented:
The angular.io preview for 1e503bf6b504cad5c1431ce43e6884022b77bd46 is available [here][1].

[1]: https://pr16228-1e503bf6b504cad5c1431ce43e6884022b77bd46.ngbuilds.io/

mary-poppins commented:
The angular.io preview for d779e8110d406febd77c152f13780c9ac876fd11 is available [here][1].

[1]: https://pr16228-d779e8110d406febd77c152f13780c9ac876fd11.ngbuilds.io/

petebacondarwin commented:
We decided that content specific images should be moved out of the `assets` folder, which should really only contain images that are general to the app shell. But I see that there is quite a mix of stuff in there right now. So instead I am going to create a post-release issue to move all that stuff around later.

mary-poppins commented:
The angular.io preview for 6c3d56193a181a059692a0bfd040fe1b7d809328 is available [here][1].

[1]: https://pr16228-6c3d56193a181a059692a0bfd040fe1b7d809328.ngbuilds.io/

mary-poppins commented:
The angular.io preview for bf50ce1a094ca8aa4c858c4cc65b792f790fa4ea is available [here][1].

[1]: https://pr16228-bf50ce1a094ca8aa4c858c4cc65b792f790fa4ea.ngbuilds.io/
